hive-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/HiveBatchSource.java

```diff
@@ -83,38 +82,33 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
 
   @Override
   public void prepareRun(BatchSourceContext context) throws Exception {
-    // This line is to load VersionInfo class here to make it available in the HCatInputFormat.setInput call. This is
-    // needed to support CDAP 3.2 where we were just exposing the classes of the plugin jar and not the resources.
-    LOG.trace("Hadoop version: {}", VersionInfo.getVersion());
-    Job job = JobUtils.createInstance();
-    Configuration conf = job.getConfiguration();
-
-    conf.set(HiveConf.ConfVars.METASTOREURIS.varname, config.metaStoreURI);
-
-    if (UserGroupInformation.isSecurityEnabled()) {
-      conf.set(HiveConf.ConfVars.METASTORE_USE_THRIFT_SASL.varname, "true");
-      conf.set("hive.metastore.token.signature", HiveAuthFactory.HS2_CLIENT_TOKEN);
-    }
-    // Use the current thread's classloader to ensure that when setInput is called it can access VersionInfo class
-    // loaded above. This is needed to support CDAP 3.2 where we were just exposing classes to plugin jars and not
-    // resources.
+    // TODO: remove after CDAP-5950 is fixed
     ClassLoader classLoader = Thread.currentThread().getContextClassLoader();
     try {
       Thread.currentThread().setContextClassLoader(getClass().getClassLoader());
```

Comment:
nit: seems like by convention, we set the class loader outside the ``try...finally``